go.mod

```diff
@@ -1,39 +1,39 @@
 module github.com/hashicorp/terraform-provider-kubernetes
 
-go 1.21
```

Comment:
what's the reason for the change go.mod and go.sum?

Reply:
@BBBmau I've reduced the number of changes to the go.mod, the main change is I've added the `github.com/hashicorp/terraform-plugin-codegen-kubernetes/`, which in turn has updated the plugin framework and other terraform dependencies.

Reply:
But isn't `github.com/hashicorp/terraform-plugin-codegen-kubernetes/autocrud` required for the CRUD operations? 

I had to also utilize it here - 

https://github.com/hashicorp/terraform-provider-kubernetes/blob/93654c59dcab99445c667b0c3e5345262fb7c8fb/internal/framework/provider/appsv1/validating_admission_policy.go#L22-L27

Reply:
ah yeah you're right. Apologies!

It's also worth noting that after reviewing the issue further that that once this PR is complete it unfortunately won't be merged until `v3.0.0` due to this resource being supported by default in k8s `1.30`. We intend to do a major version bump in order to support this since we currently only support up to `v1.28`. I've added it as part of `v3..0.0` milestone as of now.

Of course you can still work on it but wanted to let you know in case you felt that this was on a deadline. We appreciate you working on this!

Reply:
Sure, thanks for letting me know! If there are any other issues that require immediate attention I'd be glad to help out!